src/target/riscv/riscv.c

```diff
@@ -5588,6 +5588,52 @@ COMMAND_HANDLER(handle_riscv_virt2phys_mode)
 	return ERROR_OK;
 }
 
+COMMAND_HANDLER(riscv_smp_set_ext_triggers)
+{
+	struct target *target = get_current_target(CMD_CTX);
+	RISCV_INFO(r);
+
+	if (!target->smp) {
+		LOG_ERROR("Command cannot be used on a target that does not belong to any SMP group.");
+		return ERROR_COMMAND_SYNTAX_ERROR;
+	}
+
+	if (CMD_ARGC % 2) {
+		LOG_ERROR("Command takes an even number of parameters.");
+		return ERROR_COMMAND_SYNTAX_ERROR;
+	}
+
+	if (CMD_ARGC > 2 * RISCV_MAX_EXTTRIGGERS) {
+		LOG_ERROR("The number of parameters can not exceeds %d.", 2 * RISCV_MAX_EXTTRIGGERS);
+		return ERROR_COMMAND_SYNTAX_ERROR;
+	}
+
+	unsigned int index = 0;
+	for (unsigned int i = 0; i < CMD_ARGC - 1; i += 2) {
+		if (!strcmp("halt_group", CMD_ARGV[i])) {
+			r->external_triggers[index].grouptype = HALT_GROUP;
+		} else {
+			/* Ext. triggers for resume groups are not supported: neither OpenOCD nor GDB
+			 * have a concept of hart resuming on its own due to an external reason, without
+			 * an explicit resume request. */
+			LOG_ERROR("%s is not a valid argument. "
+				"The only supported group type is 'halt_group'.", CMD_ARGV[0]);
+			return ERROR_COMMAND_SYNTAX_ERROR;
```

Comment:
This would be a more fitting return code:

```suggestion
			return ERROR_COMMAND_ARGUMENT_INVALID;
```